stan/math/opencl/kernel_generator/is_valid_expression.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef STAN_MATH_OPENCL_IS_USABLE_AS_OPERATION
+#define STAN_MATH_OPENCL_IS_USABLE_AS_OPERATION
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/kernel_generator/operation.hpp>
+#include <stan/math/prim/meta.hpp>
+#include <type_traits>
+
+namespace stan {
+namespace math {
+
+/**
+ * Determines whether a type is non-scalar type that is a valid kernel generator
+ * expression.
+ */
+template <typename T>
+struct is_valid_expression_and_not_scalar {
+  enum { value = std::is_base_of<operation_base, T>::value };
+};
```

Comment:
Pretty sure you can clean these up with

```
template <typename T>
using is_valid_expression_and_not_scalar = std::is_base_of<operation_base, T>;
```

Reply:
I think we can't do this, since we need a specialization for `matrix_cl`, which must be templated itself.